sound/soc/sof/sof-priv.h

```diff
@@ -425,12 +425,15 @@ struct sof_ipc_pm_ops {
  * @query_fw_configuration: Optional function pointer to query information and
  *			configuration from the booted firmware.
  *			Executed after the first successful firmware boot.
+ * @reload_fw_addons:	Optional function pointer to reload firmware components
+ *			after the base firmware has been booted up
```

Comment:
naming nit-pick: why are we talking about 'reloading", that would mean we've loaded those component already. Did you mean 'load_fw_addons'?

and I also don't get why we can't be explicit about libraries, I don't know what 'addons' might mean, that could mean very different things to different people.

And we also need to be clear that there is no 'unload'

Reply:
Right, my idea with this was to give a point where IPC dependent post firmware boot loading, patching can be executed. Be it libraries, some blobs, tuning values, whatever.
Basically a fixed point right after firmware boot to download (if needed) anything that the firmware might need prior to continuing to configure and run.

I have chosen reload, but perhaps `restore` might be better? If it is load_fw_addons then this callback must be also called after the first boot, right? Without condition, after `query_fw_configuration()`

Reply:
@ujfalusi Isn't this quite close to existing "post_fw_run()" at DSP ops level? Maybe just have same for ipc_fw_loader ops (i.e. define post_fw_run also for fw_loaded_ops...?

Reply:
right, so it is really need to be in the fw_loader_ops. I'll do that.